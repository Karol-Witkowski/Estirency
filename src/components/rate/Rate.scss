$primary: #1ebffa;

.rate {
  font-size: 0.95em;
  color: #1cace0;
  text-transform: uppercase;
  text-align: center;
  float: right;
  width: 20%;
  height: 21.6%;
  margin-top: 22%;
  padding-left: 1.5em;
  float: left;
  display: block;
  background-color: rgb(35, 35, 35);
  box-shadow: (2px 8px 30px -15px $primary);
  height: 580px;
  border: solid 1px $primary;
  border-radius: 10px;
  margin-top: 50px;
  margin-left: 30px;

  .transactionRate {
    padding-top: 7.5%;
    margin-right: 1.6em;
    margin-left: -2%;
  }

  .dataInfo {
    font-weight: 600;
    font-size: 0.8em;
    text-align: justify;
    margin: 30px 30px 20px 0px;
  }

  .apiInfo {
    text-align: center;
    margin: 30px 30px 10px 0px;
    font-weight: 600;
    font-size: 0.8em;

    .apiIcon {
      margin-top: 30px;
      display: block;
      align-self: center;
    }
  }

  .calculated {
    font-size: 1.16em;
    color: $primary;
  }
}