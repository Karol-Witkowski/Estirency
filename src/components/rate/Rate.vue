<template>
  <section class="rate">
      <header>
        <h3>Description</h3>
      </header>
      <p>The rate of transaction is <span class="numbers">{{ setRate }}</span></p>
      <span class="numbers">{{ $store.state.amountValue }}</span>
      <span>
        {{ $store.state.baseCurrency.symbol }} ({{ $store.state.baseCurrency.name }}) is
      </span>
      <span class="numbers">{{ setAmount }}</span>
      <span>
        {{ $store.state.targetCurrency.symbol }} ({{ $store.state.targetCurrency.name }})
      </span>
      <p>
        The data is provided by the European Central Bank placed in Frankfurt am Main and
        base on The reference rates that are updated usually around<span>16:00</span>CET
        on every working day.
      </p>
      <h5 class="apiInfo">This app was created thanks to the API service provided by:</h5>
      <div class="apiIcon">
        <a
          href="https://exchangeratesapi.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <figure>
            <img
              alt="API provider logo"
              src="@/assets/api.svg"
            >
          </figure>
        </a>
      </div>
  </section>
</template>

<script>
import axios from 'axios';

export default {
  name: 'Rate',

  data() {
    return {
      rate: Number,
    };
  },

  computed: {
    setRate() {
      return Number(parseFloat(this.rate).toFixed(3));
    },

    setAmount() {
      return Number((this.$store.state.amountValue * this.setRate).toFixed(2));
    },
  },

  methods: {
    getRates() {
      const currencyRate = 'https://api.exchangeratesapi.io/latest?base=';

      axios.get(`${currencyRate}${this.$store.state.baseCurrency.cc}`)
        .then((response) => {
          this.rate = response.data.rates[this.$store.state.targetCurrency.cc];
        })
        .catch((error) => error);
    },
  },

  mounted() {
    this.getRates();
  },

  updated() {
    this.getRates();
  },
};
</script>

<style src="./Rate.scss" lang="scss"/>
