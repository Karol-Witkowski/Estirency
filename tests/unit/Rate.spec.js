import { shallowMount } from '@vue/test-utils';
import Rate from '@/components/Rate.vue';
import axios from 'axios';

let wrapper;
const currencies = [
  { cc: 'PLN', symbol: 'z\u0142', name: 'Polish zloty' },
  { cc: 'EUR', symbol: '\u20ac', name: 'European Euro' },
];

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({
    data: {
      rates: {
        EUR: 2.1234,
        PLN: 4.1234,
      },
    },
  })),
}));

beforeEach(() => {
  wrapper = shallowMount(Rate, {
    mocks: {
      $store: {
        state: {
          currency: currencies,
          baseCurrency: currencies[0],
          targetCurrency: currencies[1],
          amountValue: 2,
        },
      },
    },
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('Rate.vue test', () => {
  it('Render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Check mocked store', () => {
    expect(wrapper.find('span[name=baseAmount]').text()).toMatch('2');
    expect(wrapper.find('span[name=base]').text()).toMatch('zł (Polish zloty)');
    expect(wrapper.find('span[name=target]').text()).toMatch('€ (European Euro)');
  });

  it('Check axios call', () => {
    expect(axios.get).toHaveBeenCalled();
  });

  it('Check setRate computed', () => {
    expect(wrapper.vm.setRate).toEqual(2.123);
  });

  it('Check setAmount computed', () => {
    expect(wrapper.vm.setAmount).toEqual(4.25);
  });
});
