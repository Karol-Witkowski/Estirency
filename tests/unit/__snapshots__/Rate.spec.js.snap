// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate.vue test Render correctly 1`] = `
<section>
  <header>
    <h3>Description</h3>
  </header>
  <p>The rate of transaction is
    <span name="rate" class="numbers">
      2.123
    </span>
  </p> <span name="baseAmount" class="numbers">
      2
    </span> <span name="base">
    zł (Polish zloty) is
  </span> <span name="targetAmount" class="numbers">
    4.25
  </span> <span name="target">
    € (European Euro)
  </span>
  <p>
    The data is provided by the European Central Bank placed in Frankfurt am Main and
    base on The reference rates that are updated usually around <span>16:00</span>CET
    on every working day.
  </p>
  <h5>This app was created thanks to the API service provided by:</h5> <a href="https://exchangeratesapi.io/" rel="noopener noreferrer" target="_blank">
    <figure><img alt="API provider logo" src="@/assets/api.svg"></figure>
  </a>
</section>
`;
